bool head_to_position
---
bool finished_at_start_point